docs/src/main/asciidoc/scripting.adoc

```diff
@@ -0,0 +1,440 @@
+If you want to clear the caches to force resolution use `jbang cache clear`
+====
+
+== Using injection
+
+Dependency injection in Quarkus is based on ArC which is a CDI-based dependency injection solution tailored for Quarkus' architecture.
+You can learn more about it in the link:cdi-reference[Contexts and Dependency Injection guide].
+
+ArC comes as a dependency of `quarkus-resteasy` so you already have it handy.
+
+Let's modify the application and add a companion bean.
+
+Normally you would add a seperate class, but as we are aiming to have it all in one file you will add a 
+nested class.
+
+Add the following *inside* the `quarkusapp` class body.
+
+[source, java]
+----
+@ApplicationScoped
+static public class GreetingService {
+
+    public String greeting(String name) {
+        return "hello " + name;
+    }
+
+}
+----
+
+[TIP]
+.Use of nested static public classes
+====
+We are using a nested static public class instead of a top level class for two reasons:
+
+  1. jbang currently does not currently support multiple source files
+  2. All java frameworks relying on introspection have challenges using top level classes as they are not as visible as public classes; and in java there can only be one top level public class in a file.
+
+====
+
+Edit the `quarksapp` class to inject the `GreetingService` and create a new endpoint using it, you should end up with something like:
+
+[source, java]
+----
+//usr/bin/env jbang "$0" "$@" ; exit $?
+//DEPS io.quarkus:quarkus-resteasy:999-SNAPSHOT
+
+import io.quarkus.runtime.Quarkus;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+@Path("/hello")
+@ApplicationScoped
+public class quarkusapp {
+
+    @GET
+    public String sayHello() {
+        return "hello from Quarkus with jbang.dev";
+    }
+
+    public static void main(String[] args) {
+        Quarkus.run(args);
+    }
+
+    @Inject
+    GreetingService service;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/greeting/{name}")
+    public String greeting(@PathParam String name) {
+        return service.greeting(name);
+    }
+
+    @ApplicationScoped
+    static public class GreetingService {
+
+        public String greeting(String name) {
+            return "hello " + name;
+        }
+    }
+}
+----
+
+Now when you run `jbang quarkusapp.java` you can check what the new end point returns:
+
+[source,shell,subs=attributes+]
+----
+$ curl -w "\n" http://localhost:8080/hello/greeting/quarkus
+hello null
+----
+
+Now that is unexpected, why is it returning `hello null` and not `hello quarkus` ?
+
+The reason is that JAX-RS `@PathParam` relies on the `-parameters` compiler flag to be set to be able to map `{name}` to the `name` parameter.
```

Comment:
Only in Quarkus or is that per spec?

Reply:
Per spec. 

Openjdk didn't enable it by default for whatever reasons :)